src/dill/gauge-proxy-v2.sol

```diff
@@ -0,0 +1,2161 @@
+
+        if (_indefinite == true) {
+            _delegate.indefinite = true;
+        } else if (_delegate.prevDelegate == address(0)) {
+            _delegate.endPeriod = currentPeriodId + _periodsCount - 1;
+        } else {
+            _delegate.endPeriod = currentPeriodId + _periodsCount;
+        }
+    }
+
+    function voteFor(
+        address _owner,
+        address[] calldata _tokenVote,
+        int256[] calldata _weights
+    ) external {
+        require(
+            _tokenVote.length == _weights.length,
+            "GaugeProxy: token votes count does not match weights count"
+        );
+
+        uint256 currentId = getCurrentPeriodId();
+        require(currentId > 0, "Voting not started yet");
+        delegateData storage _delegate = delegations[_owner];
+        require(
+            (_delegate.delegate == msg.sender &&
+                currentId > _delegate.updatePeriodId) ||
+                (_delegate.prevDelegate == msg.sender &&
+                    currentId == _delegate.updatePeriodId) ||
+                (_delegate.prevDelegate == address(0) &&
+                    currentId == _delegate.updatePeriodId),
+            "Sender not authorized"
+        );
+        require(
+            _delegate.blockDelegate[currentId] == false,
+            "Delegating address has already voted"
+        );
+        require(
+            (_delegate.indefinite || currentId <= _delegate.endPeriod),
+            "Delegating period expired"
+        );
+
+        _vote(_owner, _tokenVote, _weights, currentId);
+    }
+
+    // Add new token gauge
+    function addGauge(address _token) external {
+        require(msg.sender == governance, "!gov");
+        require(
+            address(gaugeMiddleware) != address(0),
+            "cannot add new gauge without initializing gaugeMiddleware"
+        );
+        require(gauges[_token] == address(0x0), "exists");
+        string[] memory _rewardSymbols = new string[](1);
+        address[] memory _rewardTokens = new address[](1);
+        _rewardSymbols[0] = "PICKLE";
+        _rewardTokens[0] = address(PICKLE);
+        gauges[_token] = gaugeMiddleware.addGauge(
+            _token,
+            governance,
+            _rewardSymbols,
+            _rewardTokens
+        );
+        _tokens.push(_token);
+    }
+
+    // Add new token virtual gauge
+    function addVirtualGauge(address _token, address _jar) external {
+        require(msg.sender == governance, "!gov");
+        require(
+            address(virtualGaugeMiddleware) != address(0),
+            "cannot add new gauge without initializing gaugeMiddleware"
+        );
+        require(gauges[_token] == address(0x0), "exists");
+        string[] memory _rewardSymbols = new string[](1);
+        address[] memory _rewardTokens = new address[](1);
+        _rewardSymbols[0] = "PICKLE";
+        _rewardTokens[0] = address(PICKLE);
+        address vgauge = virtualGaugeMiddleware.addVirtualGauge(
```

Comment:
For consistency, `gauges[_token]` can be assigned the return value from `addVirtualGauge`, like above